examples/pipelines/fl_cross_silo_factory/submit.py

```diff
@@ -282,132 +286,71 @@ def silo_scatter_subgraph(
         # DP max gradient norm
         dp_max_grad_norm=dp_max_grad_norm,
         # Total number of iterations
-        total_num_of_iterations=total_num_of_iterations,
+        total_num_of_iterations=num_of_iterations,
         # Silo name/identifier
-        metrics_prefix=scatter_name,
+        metrics_prefix=silo_name,
         # Iteration number
         iteration_num=iteration_num,
     )
 
     # Assigning the silo's second compute to the training component
-    silo_training_step.compute = scatter_compute2
+    silo_training_step.compute = silo_compute2
 
-    # IMPORTANT: we will assume that any output provided here can be exfiltrated into the orchestrator
+    # IMPORTANT: we will assume that any output provided here can be exfiltrated into the orchestrator/gather
     return {
         # NOTE: the key you use is custom
         # a map function scatter_to_gather_map needs to be provided
         # to map the name here to the expected input from gather
-        "input_silo": silo_training_step.outputs.model
+        "model": silo_training_step.outputs.model
     }
 
 
 #######################
-### D. FACTORY CODE ###
+### D. FL Contract ###
 #######################
 
-# In this section, we're using a wrapper to build the full FL pipeline
-# based on the custom methods you implemented in section C.
-
-# 1. create an instance of the factory
-builder = FederatedLearningPipelineFactory()
-
-# 2. feed it with FL parameters
-
-builder.set_orchestrator(
-    # provide settings for orchestrator
-    YAML_CONFIG.federated_learning.orchestrator.compute,
-    YAML_CONFIG.federated_learning.orchestrator.datastore,
-)
+from fl_helper import scatter_gather
+
+scatter_configs = [
+    {
+        "inputs": {
+            "raw_train_data": Input(**dict(silo_config["inputs"])["raw_training_data"]),
+            "raw_test_data": Input(**dict(silo_config["inputs"])["raw_testing_data"]),
+        },
+        "computes": silo_config["computes"],
+        "datastore": silo_config["datastore"],
+        "name": silo_config["name"],
```

Comment:
I don't think "name" is part of our contract? Should we just inject "name" under "inputs" as a string?

Reply:
done